### PR TITLE
Small pieces of rename serialization cleanup

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fallout after moving rename to have a more lightweight form of serialization when processing.  Something i also want to pull out of a different, more complex, rename oop PR.

### CODE REVIEW

File: src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs [hunk 1]
```diff
@@ -79,12 +79,14 @@ private CodeCleanupOptionsProvider GetClientOptionsProvider(RemoteServiceCallbac
...
                     return null;
 
-                var fallbackOptions = GetClientOptionsProvider(callbackId);
-
-                var result = await LightweightRenameLocations.FindRenameLocationsAsync(
-                    symbol, solution, options, fallbackOptions, cancellationToken).ConfigureAwait(false);
-
-                return result.Dehydrate();
+                var renameLocations = await SymbolicRenameLocations.FindLocationsInCurrentProcessAsync(
+                    symbol, solution, options, GetClientOptionsProvider(callbackId), cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs:83]: there was no point jumping to LightweightRenameLocations on the *oop* side.  it exists on the host side as a temporary data so we can call back into oop efficiently without rehydrating symbols.  THe previous code would make the LightweightRenameLocations  then immediately convert that to SymbolicRenameLocations.  so this just bypasses that.

File: src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs [hunk 2]
```diff
@@ -103,12 +105,13 @@ private CodeCleanupOptionsProvider GetClientOptionsProvider(RemoteServiceCallbac
                 if (symbol is null)
                     return null;
 
-                var locations = await LightweightRenameLocations.TryRehydrateAsync(
-                    solution, GetClientOptionsProvider(callbackId), serializableLocations, cancellationToken).ConfigureAwait(false);
-                if (locations == null)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs:108]: similarly, we should not be going through LightweightRenameLocations on the oop side.

File: src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs (hunk 1 above)
Comment (ryzngard) [reply src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs:83]: Is it possible we can annotate types that shouldn't be used in OOP/Not OOP locations based on folder paths? I assume outside of serialization everything in ServiceHub/* doesn't need to use serializable data.

**Review by ryzngard — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs:83]: Definitely an good idea! :)

**Pull request merged**